hathor/transaction/storage/transaction_storage.py

```diff
@@ -338,8 +341,11 @@ def update_tx_tips(self, tx: BaseTransaction) -> None:
             # themselves become tips (hence we use to_remove_parents)
             meta = tip_tx.get_metadata()
             if meta.voided_by:
-                to_remove.add(tip_tx.hash)
-                to_remove_parents.update(tip_tx.parents)
+                if tip_tx.hash not in to_remove:
```

Comment:
The complexity in time increase considerably with these changes. It was O(n+m) and became O(n*m) in some cases.